pkg/ccl/sqlccl/csv.go

```diff
@@ -338,7 +377,30 @@ func readCSV(
 	if comma == 0 {
 		comma = ','
 	}
+
+	var totalBytes, readBytes int64
+	// Attempt to fetch total number of bytes for all files.
 	for _, dataFile := range dataFiles {
+		conf, err := storageccl.ExportStorageConfFromURI(dataFile)
+		if err != nil {
+			return 0, err
+		}
+		es, err := storageccl.MakeExportStorage(ctx, conf)
+		if err != nil {
+			return 0, err
+		}
+		sz, _ := es.Size(ctx, "")
```

Comment:
do we want to log the error? or potentially also log a warning explaining that we'll fall back in the size == 0 block below?